File: 347.top-k-frequent-elements.js
/*
 * @lc app=leetcode id=347 lang=javascript
 *
 * [347] Top K Frequent Elements
 */

// @lc code=start
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {number[]}
 */
var topKFrequent = function(nums, k) {
    let hash_dict = {};
    for (let i=0;i<nums.length;i++){
        if (hash_dict[nums[i]] !== undefined){
            hash_dict[nums[i]] += 1;
        }
        else{
            hash_dict[nums[i]] = 1;
        }
    }
    let top_k = [];
    for (let [key,value] of Object.entries(hash_dict)){
        top_k.push([value,parseInt(key)]);
    }
    top_k.sort((a,b) => {
        return b[0]-a[0];
    })
    let result = [];
    for (let i=0;i<k;i++){
        result.push(top_k[i][1]);
    }
    return result
};
// @lc code=end

File: 347.top-k-frequent-elements.py
#
# @lc app=leetcode id=347 lang=python3
#
# [347] Top K Frequent Elements
#

# @lc code=start
import heapq
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        count = {}
        freq = [[] for i in range(len(nums)+1)]
        # print(freq)
        for num in nums:
            count[num] = 1 + count.get(num,0)
        for key,value in count.items():
            freq[value].append(key)
        result = []
        # print(freq)
        for i in range(len(freq)-1,0,-1):
            for n in freq[i]:
                result.append(n)
                if len(result) == k:
                    return result
# @lc code=end

